demo/admin/src/predefinedPage/EditPredefinedPage.tsx

```diff
@@ -34,42 +36,64 @@ const getQuery = gql`
     }
 `;
 
-const updateMutation = gql`
-    mutation UpdatePredefinedPage($pageId: ID!, $input: PredefinedPageInput!, $lastUpdatedAt: DateTime, $attachedPageTreeNodeId: ID!) {
-        savePredefinedPage(id: $pageId, input: $input, lastUpdatedAt: $lastUpdatedAt, attachedPageTreeNodeId: $attachedPageTreeNodeId) {
+const savePredefinedPageMutation = gql`
+    mutation SavePredefinedPage($id: ID!, $input: PredefinedPageInput!, $lastUpdatedAt: DateTime, $attachedPageTreeNodeId: ID!) {
+        savePredefinedPage(id: $id, input: $input, lastUpdatedAt: $lastUpdatedAt, attachedPageTreeNodeId: $attachedPageTreeNodeId) {
             id
             type
         }
     }
 `;
 
+type FormValues = {
+    type?: string;
+};
+
 interface Props {
     id: string;
 }
 
-export const EditPredefinedPage = ({ id }: Props) => {
+export const EditPredefinedPage = ({ id: pageTreeNodeId }: Props) => {
     const stackApi = useStackApi();
+    const client = useApolloClient();
 
     const { data, loading } = useQuery<GQLPredefinedPageQuery, GQLPredefinedPageQueryVariables>(getQuery, {
-        variables: { id },
+        variables: { pageTreeNodeId },
     });
 
-    const [mutation] = useMutation<GQLUpdatePredefinedPageMutation, GQLUpdatePredefinedPageMutationVariables>(updateMutation, {
-        refetchQueries: ["PredefinedPage"],
-    });
+    const initialValues = useMemo<Partial<FormValues>>(() => {
+        if (data?.pageTreeNode?.document) {
+            if (data.pageTreeNode.document.__typename !== "PredefinedPage") {
+                throw new Error(`Invalid document type: ${data.pageTreeNode.document.__typename}`);
+            }
+
+            return {
+                type: data.pageTreeNode.document.type ?? undefined,
```

Comment:
isn't `?? undefined` a NOOP?

Reply:
No, if type is `null` than it will evaluate to `undefined`:

<img width="172" alt="image" src="https://github.com/user-attachments/assets/3750c85f-a36f-46b0-bedf-fbe3e68eb29a">

I need this to have the correct form values type here: https://github.com/vivid-planet/comet/pull/2550/files#diff-3c7454a62a63fcfe7e9bc99132bcfd2d9032b35597be88c9393d2416979f20ecR31-R33. Alternatively, I could change it to `type?: string | null`, but I'm not sure if this would correctly with the select.